app/models/validations/household_validations.rb

```diff
@@ -51,17 +51,114 @@ def validate_armed_forces(record)
+    (2..8).each do |person_num|
+      age = record.public_send("age#{person_num}")
+      relationship = record.public_send("relat#{person_num}")
+      next unless age && relationship
+
+      if age < 16 && !relationship_is_child_other_or_refused?(relationship)
+        record.errors.add "relat#{person_num}", I18n.t("validations.household.relat.child_under_16_lettings", person_num:)
+        record.errors.add "age#{person_num}", I18n.t("validations.household.age.child_under_16_relat_lettings", person_num:)
+      end
+
+      if record.form.start_year_after_2024? && (age > 19 && tenant_is_child?(relationship))
+        record.errors.add "relat#{person_num}", I18n.t("validations.household.relat.child_over_19", person_num:)
+        record.errors.add "age#{person_num}", I18n.t("validations.household.age.child_over_19_relat", person_num:, person: "lead tenant")
+      end
+    end
+  end
+
+  def validate_person_age_and_relationship_matches_economic_status(record)
+    (2..8).each do |person_num|
+      age = record.public_send("age#{person_num}")
+      economic_status = record.public_send("ecstat#{person_num}")
+      relationship = record.public_send("relat#{person_num}")
+      next unless age && economic_status && relationship
+
+      age_between_16_19 = age.between?(16, 19)
+      student = tenant_is_fulltime_student?(economic_status)
+      economic_status_refused = tenant_economic_status_refused?(economic_status)
+      child = tenant_is_child?(relationship)
+
+      if age_between_16_19 && !(student || economic_status_refused) && child
+        if record.form.start_year_after_2024?
+          record.errors.add "ecstat#{person_num}", I18n.t("validations.household.ecstat.student_16_19.must_be_student_2024", person_num:, person: "lead tenant")
+        else
+          record.errors.add "ecstat#{person_num}", I18n.t("validations.household.ecstat.student_16_19.must_be_student")
+        end
+        record.errors.add "age#{person_num}", I18n.t("validations.household.age.student_16_19.cannot_be_16_19.child_not_student")
+        record.errors.add "relat#{person_num}", I18n.t("validations.household.relat.student_16_19.cannot_be_child.16_19_not_student")
+      end
+
+      if !age_between_16_19 && student && child
+        if record.form.start_year_after_2024?
+          record.errors.add "age#{person_num}", I18n.t("validations.household.age.student_16_19.must_be_16_19_2024", person_num:, person: "lead tenant")
+        else
+          record.errors.add "age#{person_num}", I18n.t("validations.household.age.student_16_19.must_be_16_19")
+        end
+        record.errors.add "ecstat#{person_num}", I18n.t("validations.household.ecstat.student_16_19.cannot_be_student.child_not_16_19")
+        record.errors.add "relat#{person_num}", I18n.t("validations.household.relat.student_16_19.cannot_be_child.student_not_16_19")
+      end
+
+      next unless student && age_between_16_19 && !child && record.form.start_year_after_2024?
```

Comment:
I know it's in the ticket, but is this actually a sensible validation? For the same reasons we allow "other" as a relationship for under 16s - could be a child of a non-lead tenant but not the lead tenant (or presumably in this age range you could e.g. move somewhere with a partner while being a student).

Not objecting to it as is necessarily (think it's up to the stats team) but seems worth querying.

Reply:
Will check!